src/openrct2/ui/DummyUiContext.cpp

```diff
@@ -190,7 +190,7 @@ namespace OpenRCT2::Ui
         // In-game UI
         IWindowManager* GetWindowManager() override
```

Comment:
`std::unique_ptr::get` should never be exposed outside functions. To prevent the pointer from being stored and used elsewhere, return a reference to the `std::unique_ptr` instead. You'd need to change this in `UiContext.h` too.

```suggestion
        std::unique_ptr<IWindowManager>& GetWindowManager() override
```

Reply:
It is pretty standard practice to return the pointer and not a unique ptr ref and we do do this in other bits of the code.

Reply:
Yeah I don't think I've seen unique pointer refs before for accessors. Chromium also uses raw pointers in that case https://source.chromium.org/chromium/chromium/src/+/main:content/browser/renderer_host/render_frame_host_impl.cc;l\=2142?q=render_frame_host&ss=chromium.
The assumption is that when a raw pointer is returned, that already indicates lack of ownership. Otherwise the pointer should've been moved or released.

Reply:
I don't see an advantage of returning raw pointers over `std::unique_ptr` references (or even better `T&` if you know it's not null and not going to change). With raw pointers you don't know the lifetime and can accidentally move  or release them, both of which will not be easily done by accident if you return `std::unique_ptr<T>&`. There is also no performance overhead whatsoever (unlike with shared_ptr).

Reply:
Well the issue is you could call release on a unique ptr ref. https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rr-ptr core guidelines also agrees that the standard is a raw pointer is not owning.

Reply:
`T&` does look good to me, though. Will have to confirm later whether I can use that instead.

Reply:
Since this PR moves more towards these guidelines, I guess it's fine. I won't block this PR just for this, since the rest looks fine to me. I'll leave it to you to decide if you want to change this.

Reply:
Hm, maybe not in this PR. It looks like it'll cause a ripple effect throughout the codebase because `->` would have to be changed to `.` as well:
![Screenshot showing all the places that use the GetWindowManager accessor](https://user-images.githubusercontent.com/7199958/181414778-8f97bfda-3371-402e-866b-b1ea4958153d.png)